src/operator/controllers/intents_reconcilers/database/database_reconciler.go

```diff
@@ -169,6 +237,31 @@ func (r *DatabaseReconciler) getClusterID(ctx context.Context) (string, error) {
 	return clusterID, nil
 }
 
+func (r *DatabaseReconciler) annotateDatabaseAndUsernameOnPod(ctx context.Context, intents otterizev1alpha3.ClientIntents, username string, dbInstance string) error {
+	pod, err := r.serviceIdResolver.ResolveClientIntentToPod(ctx, intents)
```

Comment:
what about when a pod goes up after the intents already exist? will this work?

Reply:
another thought - what if there's more than one pod matching the clientintents? We will only annotate the first one we found. 

In general I think we should consider going the other way about it - using the intent-operator's pods reconciler, for every pod that has clientintents of type DB, annotate it with the db user & password. I.e., do this in the pod reconciler loop, and not in the intents reconciler loop. WDYT?

Reply:
another thought - what if there's more than one pod matching the clientintents? We will only annotate the first one we found. 

In general I think we should consider going the other way about it - using the intent-operator's pods reconciler, for every pod that has clientintents of type DB, annotate it with the db user & password. I.e., do this in the pod reconciler loop, and not in the intents reconciler loop. @evyatarmeged WDYT?

Reply:
I think we should just log a warning in case no pods exist, and call the intents reconciler from the pods reconciler (like we do in `IAMPodReconciler`) for pods with DB intents. At least until it all moves to SEP.
@amitlicht thoughts ?